src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -531,6 +533,46 @@ public async Task RestoreDatabase()
                 if (ResourceNameValidator.IsValidResourceName(databaseName, ServerStore.Configuration.Core.DataDirectory.FullPath, out string errorMessage) == false)
                     throw new BadRequestException(errorMessage);
 
+                var extension = Path.GetExtension(restoreConfigurationJson.LastFileNameToRestore);
+                if (extension == Constants.Documents.PeriodicBackup.SnapshotExtension || extension == Constants.Documents.PeriodicBackup.EncryptedSnapshotExtension)
+                {
+                    long backupSizeInBytes;
+                    var zipPath = Path.Combine(restoreConfigurationJson.BackupLocation, restoreConfigurationJson.LastFileNameToRestore);
+
+                    try
+                    {
+                        using (var zip = ZipFile.OpenRead(zipPath))
+                            backupSizeInBytes = zip.Entries.Sum(entry => entry.Length);
+                    }
+                    catch (Exception e)
+                    {
+                        if (e is InvalidDataException)
+                            throw new InvalidDataException($"Invalid snapshot file: {restoreConfigurationJson.LastFileNameToRestore} at path : {zipPath}");
+                        if (e is FileNotFoundException)
+                            throw new FileNotFoundException($"Could not find file: {restoreConfigurationJson.LastFileNameToRestore} at path: {zipPath}");
+
+                        throw new IOException($"Error reading snapshot file: {restoreConfigurationJson.LastFileNameToRestore}, at path: {zipPath}. Please provide a valid snapshot file.");
+                    }
+
+                    var baseDataDirectory = ServerStore.Configuration.Core.DataDirectory.FullPath;
+
+                    var destinationPath = string.IsNullOrEmpty(restoreConfigurationJson.DataDirectory) == false 
+                        ? PathUtil.ToFullPath(restoreConfigurationJson.DataDirectory, baseDataDirectory) 
```

Comment:
can we use `new PathSetting(restoreConfigurationJson.DataDirectory, baseDataDirectory).FullPath` here?

We do some validation inside that class.

Reply:
done